clang/include/clang/Sema/Sema.h

```diff
@@ -12455,6 +12455,7 @@ class Sema final {
   };
 
   bool isKnownGoodSYCLDecl(const Decl *D);
+  void checkSYCLVarDeclIfInKernel(VarDecl *Var);
```

Comment:
Still kind of hate the name, seems clumsy, but I don't have a better suggestion at the moment, I'll think about it.

Reply:
```suggestion
  void checkSYCLVarDeclInSYCLKernel(VarDecl *Var);
```
since SYCL is about the kernel, not the variable?

Reply:
SYCL is not only about kernels. Let's call it "device code".
```suggestion
  void diagnoseVarDeclIfSYCLDeviceCode(VarDecl *Var);
```

Reply:
Ping?

Reply:
How about 'checkSYCLDeviceVarDecl'.  "check" is the common word for a function that diagnoses.  The word "if" is incorrect, since this isn't checking whether it is in a SYCL device.

The above also is more consistent in the cases we've added where we recursively check.

Reply:
Hmm, okay. Let it be `checkSYCLDeviceVarDecl`.

Reply:
done.